giraffe/src/components/geo/processing/mixins.ts

```diff
@@ -0,0 +1,57 @@
+import {S2} from 's2-geometry'
+import {CoordinateEncoding} from './GeoTable'
+import {LatLon} from './GeoTable'
+import {
+  LAT_COLUMN,
+  LON_COLUMN,
+  START_COLUMN,
+  STOP_COLUMN,
+  TIME_COLUMN,
+} from './tableProcessing'
+
+export const getLatLonMixin = function(index: number): LatLon {
+  if (this.coordinateEncoding === CoordinateEncoding.FIELDS) {
+    return {
+      lon: this.getValue(index, LON_COLUMN),
+      lat: this.getValue(index, LAT_COLUMN),
+    }
+  } else {
+    const cellId = this.getS2CellID(index)
+    if (cellId === null || cellId.length > 16) {
```

Comment:
let's pull this `16` into a variable or leave a comment explaining why 16 is important